src/alto-ui/Form/DatePicker/DatePicker.js

```diff
@@ -84,14 +85,15 @@ class DatePicker extends React.Component {
       this.props.inputProps.onBlur(e);
     }
 
-    const date = [format, displayFormat]
-      .reduce((d, f) => d && d.isValid ? d : DateTime.fromFormat(value, f), null);
-    if (date) {
+    const date = [format, displayFormat].reduce(
+      (d, f) => (d && d.isValid ? d : DateTime.fromFormat(value, f)),
+      null
+    );
+    if (date && date.isValid) {
```

Comment:
date.isValid not necessary, if you check your const date you already have the condition inside 

(d && d.isValid)